ctru-rs/examples/file-explorer.rs

```diff
@@ -85,34 +115,49 @@ impl<'a> FileExplorer<'a> {
                     println!("{:2} - {}", i, entry.file_name().to_string_lossy());
                     self.entries.push(entry);
 
-                    // Paginate the output
                     if (i + 1) % 20 == 0 {
-                        println!("Press A to go to next page, or Start to exit");
-
-                        while self.apt.main_loop() {
-                            self.hid.scan_input();
-                            let input = self.hid.keys_down();
-
-                            if input.contains(KeyPad::KEY_A) {
-                                break;
-                            }
-
-                            if input.contains(KeyPad::KEY_START) {
-                                self.running = false;
-                                return;
-                            }
-
-                            self.gfx.wait_for_vblank();
-                        }
+                        self.wait_for_page_down();
                     }
                 }
                 Err(e) => {
                     println!("{} - Error: {}", i, e);
                 }
             }
         }
+    }
 
-        println!("Start to exit, A to select an entry by number, B to go up a directory, X to set the path.");
+    fn print_file_contents(&mut self) {
+        match std::fs::read_to_string(&self.path) {
+            Ok(contents) => {
+                println!("File contents:\n{0:->80}", "");
+                println!("{contents}");
```

Comment:
Interestingly, for a really long file this takes a noticeably long amount of time (30 seconds on my `/retroarch/retroarch.cfg`), during which you see the lines scroll past pretty quickly. I wonder if the libctru console implementation is slow or something?

Reply:
How big is that file, out of curiosity? I only tested with this small one but anything on the order of 30s seems worth investigating. In particular I would not expect romfs to be slow, but I haven't looked too close at the io implementations for it...

Reply:
It's 104 kB. I think it's just the default config, but here is the exact file (compressed so GitHub accepts it):
[retroarch_config.zip](https://github.com/Meziu/ctru-rs/files/8057564/retroarch_config.zip)

Reply:
Yep, it looks like that they _really_ needed a scroll effect, but since the console  cooperative thread doesn’t yield the scroll blocks the whole app, and takes quite a while, too.

Reply:
Oh, the scroll effect is semi-intentional? I guess fixing that would require libctru changes...